pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -135,6 +132,31 @@ func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
 	}
 	log.Infof("Namespace controller started")
 	go nc.queue.Run(stopCh)
+	go nc.startCaBundleWatcher(stopCh)
+}
+
+// startCaBundleWatcher listens for updates to the CA bundle and update cm in each namespace
+func (nc *NamespaceController) startCaBundleWatcher(stop <-chan struct{}) {
+	id, watchCh := nc.caBundleWatcher.AddWatcher()
+	defer nc.caBundleWatcher.RemoveWatcher(id)
```

Comment:
Why do we need the remove watch now? Is it in case we lose leader election, we stop ?

Reply:
Yes, to prevent leak, not only leader election, but also multi-cluster cleanup